src/screens/augmentedReality/ARShowScreen.js

```diff
@@ -112,20 +116,69 @@ export const ARShowScreen = ({ navigation, route }) => {
   );
 };
 
+/**
+ * index creation function for model and texture
+ *
+ * @param {string} startDate           start date of the model
+ * @param {number} timePeriodInDays    time period in days of the model
+ * @param {array}  scenes              array of models
+ *
+ * @return {object} both parsed values as an object, like { modalIndex: 1, textureIndex: 1 }
+ */
+const multipleSceneIndexGenerator = async ({ startDate, timePeriodInDays, scenes }) =>
+  new Promise((resolve) => {
+    let modalIndex, textureIndex;
+
+    /* all models must have the same number of variable textures. Therefore, in order to obtain the 
+       number of textures, the textures of the first model were calculated. */
+    if (scenes.length > 1) {
+      const today = new Date(),
+        differenceDate = extendedMoment.range(startDate, today).diff('days'),
+        textureCount = scenes[0]?.localUris?.filter(
+          ({ type, stable }) => !stable && type === 'texture'
+        )?.length,
+        texture = Math.floor(differenceDate / timePeriodInDays),
+        modalCount = scenes.length,
+        modal = Math.floor(texture / textureCount);
+
+      modalIndex = modal % modalCount;
+      textureIndex = texture % textureCount;
+    }
+
+    resolve({ modalIndex, textureIndex });
+  });
+
 const objectParser = async ({ item, setObject, setIsLoading, onPress }) => {
-  const parsedObject = { texture: [] };
+  const parsedObject = { textures: [] };
+  const variableTextures = [];
+
+  const { modalIndex, textureIndex } = await multipleSceneIndexGenerator({
+    startDate: item?.payload?.startDate,
+    timePeriodInDays: item?.payload?.timePeriodInDays,
+    scenes: item?.payload?.scenes
+  });
+
+  const localUris =
+    item?.payload?.scenes?.[modalIndex]?.localUris || item?.payload?.scenes?.[0]?.localUris;
+
+  localUris.filter(
+    ({ stable, uri, type }) => stable && type === 'texture' && parsedObject.textures.push({ uri })
+  );
 
-  // TODO: in the future the index 0 used here will be changed according to time logic!
-  const localUris = item?.payload?.scenes?.[0]?.localUris;
+  // combine stable textures with variable textures
+  if (typeof modalIndex === 'number' && typeof textureIndex === 'number') {
+    localUris.filter(
+      ({ stable, uri, type }) => !stable && type === 'texture' && variableTextures.push({ uri })
+    );
+    parsedObject.textures.push(variableTextures[textureIndex || 0] || []);
```

Comment:
what does `variableTextures[textureIndex || 0] || []` stand for. can you give it a name and make it a const for example?

Reply:
Yes I can. 👍

`variableTextures[textureIndex || 0] || []`
- If there is a `textureIndex` number from the `multipleSceneIndexGenerator` function, the directory where this number is located is selected.
- Otherwise, the data at index 0 is added.
- If there is none, empty array is added.

In this way, a control is provided for variable textures.